docs/book/user-guide/advanced-guide/data-management/late-materialization.md

```diff
@@ -0,0 +1,138 @@
+
+We identify two major use-cases for exchange artifacts between pipelines:
+* You semantically group your data products using [ZenML Models](model-management.md#linking-artifacts-to-models)
+* You prefer to use [ZenML Client](../../reference/python-client.md#client-methods) to bring all the pieces together
+
+In the sections below we will dive deeper into these use-cases from pipelines point of view.
+
+## Use ZenML Models to exchange artifacts
+
+The ZenML Model is an entity introduced by The Model Control Plane feature. The Model Control Plane is how you manage your models through this unified interface. It allows you to combine the logic of your pipelines, artifacts and crucial business data along with the actual 'technical model'.
+
+[ZenML Models documentation](model-management.md#linking-artifacts-to-models) describes in great details how you can link various artifacts produced within pipelines to the model and here we would focus more on the consumption aspect of it.
+
+First let's have a look at the two pipeline project, where the first pipeline is running training logic and the second runs batch inference leveraging trained model artifact:
+
+```python
+from typing_extensions import Annotated
+from zenml import get_pipeline_context, pipeline, Model
+from zenml.enums import ModelStages
+import pandas as pd
+from sklearn.base import ClassifierMixin
+
+
+@step
+def predict(
+    model: ClassifierMixin,
+    data: pd.DataFrame,
+) -> Annotated[pd.Series, "predictions"]:
+    predictions = pd.Series(model.predict(data))
+    return predictions
+
+@pipeline(
+    model=Model(
+        name="iris_classifier",
+        # Using the production stage
+        version=ModelStages.PRODUCTION,
+    ),
+)
+def do_predictions():
+    # model name and version are derived from pipeline context
+    model = get_pipeline_context().model
+    inference_data = load_data()
+    predict(
+        # Here, we load in the `trained_model` from a trainer step
+        model=model.get_model_artifact("trained_model"),  
+        data=inference_data,
+    )
+
+
+if __name__ == "__main__":
+    do_predictions()
+```
+
+In the example above we used `get_pipeline_context().model` property to acquire the model context pipeline is running in. During pipeline compilation this context will be not evaluated, because `Production` model version is not stable version name and another model version can become `Production` before it comes to actual step execution. Same applies to calls like `model.get_model_artifact("trained_model")` - it will get stored in step configuration for late materialization during step run only.
```

Comment:
```suggestion
In the example above we used `get_pipeline_context().model` property to acquire the model context in which the pipeline is running. During pipeline compilation this context will not yet have been evaluated, because `Production` model version is not a stable version name and another model version can become `Production` before it comes to the actual step execution. The same applies to calls like `model.get_model_artifact("trained_model")`; it will get stored in the step configuration for delayed materialization which will only happen during the step run itself.
```